### PR TITLE
jax.numpy: better docstring for iscomplexobj function

### PR DESCRIPTION
**Pull request by selamw1**
New docstring and JAX-specific code examples are addd for jax.numpy.iscomplexobj.
Part of [21461](https://github.com/google/jax/issues/21461)

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -127,8 +127,35 @@ def canonicalize_shape(shape: Any, context: str="") -> core.Shape:
 printoptions = np.printoptions
 set_printoptions = np.set_printoptions
 
-@util.implements(np.iscomplexobj)
 def iscomplexobj(x: Any) -> bool:
+  """Checks if the input is a complex number or an array containing complex elements.
+
+  JAX implementation of :func:`numpy.iscomplexobj`.
+
+  The function evaluates based on input type rather than value. Inputs with zero imaginary parts are still considered complex.
```
Comment (dfm) [inline jax/_src/numpy/lax_numpy.py:135]: We're not very strict about this, but we typically want to wrap to ~80 character lines.

**Review by dfm — CHANGES_REQUESTED** [review]
Thanks! I'll let @jakevdp take a closer look when he gets a chance, but I noted some tiny changes in the meantime.

**Comment by selamw1** [comment]
Thanks @dfm 
Long characters wrapped and commits squashed into one.

**Review by jakevdp — APPROVED** [review]
Looks good, thanks!

File: jax/_src/numpy/lax_numpy.py [hunk 2]
```diff
@@ -1159,6 +1188,36 @@ def gradient_along_axis(a, h, axis):
 
 @util.implements(np.isrealobj)
 def isrealobj(x: Any) -> bool:
+  """Check if the input is a complex number or an array containing complex elements.
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:1191]: This looks like a copy of the `iscomplexobj` docstring. It will need to be updated.

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/lax_numpy.py (hunk 2 above)
Comment (selamw1) [reply jax/_src/numpy/lax_numpy.py]: You are right unfinished  `isrealobj` was commited while fixing `iscomplexobj` lint white space. Its updated now.

**Review by jakevdp — APPROVED** [review]
Thanks!

**Comment by jakevdp** [comment]
Looks like there are lint issues related to trailing whitespace

**Pull request merged**